### PR TITLE
[add] Astro.props added to config

### PR DESCRIPTION
**Pull request by sarah11918**
- New or updated content

Addresses #632 

Just getting something super simple started!

### CODE REVIEW

File: src/pages/en/reference/api-reference.md [hunk 1]
```diff
@@ -72,6 +72,29 @@ const posts = await Astro.glob<Frontmatter>('../pages/post/*.md');
...
+```
+
+```astro
+---
+// ./src/pages/index.astro
+import Heading from '../components/Heading.astro';
+---
+<Heading title="My First Post" date="09 Aug 2022" />
+```
+
```
Comment (delucis) [inline src/pages/en/reference/api-reference.md:97]: Noticing we don’t seem to do this elsewhere on this page but these links could be helpful?

```suggestion

📚 Learn more about how [Markdown Layouts](/en/guides/markdown-content/#markdown-layouts) handle props.

📚 Learn how to add [Typescript type definitions for your props](/en/guides/typescript/#component-props).

```

**Review by delucis — COMMENTED** [review]
I'd already forgotten we needed this 😅 — thankfully we have @sarah11918!

I’ve left a few comments but honestly LGTM!

File: src/pages/en/reference/api-reference.md [hunk 2]
```diff
@@ -72,6 +72,29 @@ const posts = await Astro.glob<Frontmatter>('../pages/post/*.md');
 </ul>
 ```
 
+### `Astro.props`
+
+`Astro.props` is an object containing any values that have been passed through component attributes, or through the `layout:` property in `.md` and `.mdx` files.
```
[the remark this answers is not in the document]
Comment (sarah11918) [reply src/pages/en/reference/api-reference.md:77]: I don't know why I should be allowed to edit YOUR comments, but I just did, to make the internal link!  😅

**Comment by sarah11918** [comment]
All cleaned up for @bholmesdev!

**Comment by kylebutts** [comment]
Hi @sarah11918, it might be helpful to include an example of how to set default values:

```astro
const { title, date = String(today.getDate()) } = Astro.props;
```

**Comment by delucis** [comment]
> it might be helpful to include an example of how to set default values

Agreed! But maybe that should go on the 'Astro Components' page which has a more detailed section on using component props. What do you think?

**Comment by sarah11918** [comment]
@kylebutts @delucis Agreed, this seems more like something for the components page to me. I think it would go nicely after we show using TypeScript with your props, as a 3rd example here:  https://docs.astro.build/en/core-concepts/astro-components/#component-props

@kylebutts, are you interested in making a PR for a 3rd example, following the model of the other two code examples? (We have some specific ways of authoring code samples that I don't have a guide to yet, but we can polish in a PR if we have to).

File: src/pages/en/reference/api-reference.md [hunk 3]
```diff
@@ -80,6 +80,33 @@ const posts = await Astro.glob<Frontmatter>('../pages/post/*.md');
...
+
+```astro /title=".+"/ /date=".+"/
+---
+// ./src/pages/index.astro
+import Heading from '../components/Heading.astro';
+---
+<Heading title="My First Post" date="09 Aug 2022" />
+```
+
+📚 Learn more about how [Markdown Layouts](/en/guides/markdown-content/#markdown-layouts) handle props.
```
Comment (sarah11918) [inline src/pages/en/reference/api-reference.md:106]:
```suggestion
📚 Learn more about how [Markdown and MDX Layouts](/en/guides/markdown-content/#markdown-and-mdx-layouts) handle props.
```

File: src/pages/en/reference/api-reference.md [hunk 4]
```diff
@@ -80,6 +80,33 @@ const posts = await Astro.glob<Frontmatter>('../pages/post/*.md');
...
+
+```astro /title=".+"/ /date=".+"/
+---
+// ./src/pages/index.astro
+import Heading from '../components/Heading.astro';
+---
+<Heading title="My First Post" date="09 Aug 2022" />
+```
+
+📚 Learn more about how [Markdown and MDX Layouts](/en/guides/markdown-content/#markdown-and-mdx-layouts) handle props.
```
Comment (sarah11918) [inline src/pages/en/reference/api-reference.md:106]:
```suggestion
📚 Learn more about how [Markdown and MDX Layouts](/en/guides/markdown-content/#frontmatter-layout) handle props.
```

**Pull request merged**